packages/backend/src/server/FileServerService.ts

```diff
@@ -527,15 +527,27 @@ export class FileServerService {
 		const isWebpublic = file.webpublicAccessKey === key;
 
 		if (!file.storedInternal) {
-			if (!(file.isLink && file.uri)) return '204';
-			const result = await this.downloadAndDetectTypeFromUrl(file.uri);
-			return {
-				...result,
-				url: file.uri,
-				fileRole: isThumbnail ? 'thumbnail' : isWebpublic ? 'webpublic' : 'original',
-				file,
-				filename: file.name,
-			};
+			if (file.isLink && file.uri) {
+				const result = await this.downloadAndDetectTypeFromUrl(file.uri);
+				return {
+					...result,
+					url: file.uri,
+					fileRole: isThumbnail ? 'thumbnail' : isWebpublic ? 'webpublic' : 'original',
+					file,
+					filename: file.name,
+				};
+			} else if (file.createdByMicropub) {
```

Comment:
Is there a reason for code duplication here? Why not put the `file.createdByMicropub` condition as a logical OR to the `file.isLink && file.uri`?

Reply:
Ah never mind, I didn't look closely at the code. file.ur**i** != file.ur**l**